soracom-inventory-agent-for-java-core/src/main/java/io/soracom/inventory/agent/core/initialize/InventoryAgentInitializer.java

```diff
@@ -285,15 +285,15 @@ private Credentials loadCredentials() {
 				log.info("load credentials from credential store.");
 			}
 		}
-		credentials.setLifetime(calculateLifetimeSec());
+		setRecommendedLifetime(credentials);
 		return credentials;
 	}
 
-	private long calculateLifetimeSec() {
-		int defaultLifetimeSec = 60;
-		// according to
-		// org.eclipse.leshan.client.californium.CaliforniumEndpointsManager.java#getMaxCommunicationPeriodFor
-		return defaultLifetimeSec + 247 + 30;
+	private void setRecommendedLifetime(Credentials credentials) {
+		if (credentials != null) {
+			long recommendedLifetime = InventoryAgentHelper.calculateRecommendedLifetimeSec(credentials.getLifetime());
+			credentials.setLifetime(recommendedLifetime);
+		}
```

Comment:
It seems this makes sense, but once the `credentials` was null, there is no chance to set the lifetime property?

Reply:
Yes, if the credentials is null, bootstrap is happen to get credentials from bootstrap server. The return value of the bootstrap request contains default lifetime value, and the client tries to set recommended lifetime value after bootstrap succeeds.
https://github.com/soracom/soracom-inventory-agent-for-java/pull/14/files/706928bb77e5ab315e28df642835f58343c3a3e2#diff-0185eee4d952255fa6beb44a27fb284b091a5e83aa4c9e8e170591c07c345a9cR76